### PR TITLE
refactor: browser flow

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary
This PR reorganize browser flow with an `AuthManager` which make our sign in/out process more concise.

## Documentation
[Logto Android SDK](https://www.notion.so/silverhand/Logto-Android-SDK-74a7f0abd5f34848944b1f8dd59127d3)

### CODE REVIEW

File: android/library/src/main/java/io/logto/android/Logto.kt
Comment (gao-sun) [inline android/library/src/main/java/io/logto/android/Logto.kt:41]: thought `AuthManager` will be a high level abstract instead of an additional layer.. let's go through the docs first

Comment (xiaoyijun) [reply android/library/src/main/java/io/logto/android/Logto.kt:34]: TBD: [LOG-149](https://linear.app/silverhand/issue/LOG-149/research-best-practice-in-browser-flow)

File: android/library/src/main/java/io/logto/android/storage/CredentialStorage.kt
Comment (gao-sun) [inline android/library/src/main/java/io/logto/android/storage/CredentialStorage.kt:13]: this should be more specific, add a linear issue to get persist key hash for config and think about how it works when user tries to create for a same config twice (throw an error maybe)

**Review by gao-sun — APPROVED** [review]
approve to unblock, please use squash merge and add linear issue before merging

File: android/library/src/main/java/io/logto/android/storage/CredentialStorage.kt
Comment (xiaoyijun) [reply android/library/src/main/java/io/logto/android/storage/CredentialStorage.kt:13]: TBD: 
[LOG-151](https://linear.app/silverhand/issue/LOG-151/enhance-process-correctly-when-use-the-same-logtoconfig-to-instantiate)
[LOG-150](https://linear.app/silverhand/issue/LOG-150/improvement-use-hash-to-make-logto-config-key-unique)

**Pull request merged**
